middlewares/accesslog/logger.go

```diff
@@ -83,6 +92,18 @@ func NewLogHandler(config *types.AccessLog) (*LogHandler, error) {
 		}
 	}
 
+	if config.BufferingSize > 0 {
+		go func() {
+			for {
```

Comment:
WDYT about moving the for/range here?

Reply:
If you move it here, don't you need to wait until goroutine end to close file (with WaitGroup per example)?

Reply:
You're right, you need to add a WaitGroup in order to be sure that Close wait for the drain of logs.

In addition, FYI `ok==true` until the chan is empty, so with this code, you will still loop until the chan is empty.

Reply:
I have moved for loop range to goroutine and add sync.WaitGroup to ensure stream is empty before closed file.